### PR TITLE
Better error handling

### PR DESCRIPTION
**Pull request by maxclaus**
The previous approach was too generic and was causing (at least for me)
some misunderstanding. Because even existing my “auto_updater.json” file
I was receiving that generic error message. And the real problem was
occurring while parsing the `res.body` because there was an extra comma
in my JSON file.

### CODE REVIEW

File: src/GhReleases.js [hunk 1]
```diff
@@ -70,19 +71,32 @@ export default class GhReleases extends events.EventEmitter {
     // Make sure feedUrl exists
     return got.get(feedUrl)
       .then(res => {
-        if (res.statusCode !== 200) throw new Error()
+        if (res.statusCode === 404) {
+          throw new Error('auto_updater.json does not exist or does not links to the latest GitHub release.')
```
Comment (jenslind) [inline src/GhReleases.js]: The error message here should be only: `auto_updater.json does not exist.`

File: src/GhReleases.js [hunk 2]
```diff
@@ -70,19 +71,32 @@ export default class GhReleases extends events.EventEmitter {
...
-        let zipUrl = JSON.parse(res.body).url
-        if (semver.clean(zipUrl.split('/').slice(-2, -1)[0]) !== semver.clean(tag)) {
-          throw new Error()
+        const matchReleaseUrl = zipUrl.match(REGEX_ZIP_URL)
+        if (!matchReleaseUrl) {
+          throw new Error('The zipUrl (' + versionInZipUrl + ') is a invalid release URL')
+        }
+
+        const versionInZipUrl = matchReleaseUrl[1]
+        const currentVersion = semver.clean(tag);
```
Comment (jenslind) [inline src/GhReleases.js]: currentVersion => latestVersion

File: src/GhReleases.js [hunk 3]
```diff
@@ -70,19 +71,32 @@ export default class GhReleases extends events.EventEmitter {
...
+          throw new Error('Unable to parse the auto_updater.json: ' + err.message + ', body: ' + res.body)
+        }
 
-        // Make sure the feedUrl links to latest tag
-        let zipUrl = JSON.parse(res.body).url
-        if (semver.clean(zipUrl.split('/').slice(-2, -1)[0]) !== semver.clean(tag)) {
-          throw new Error()
+        const matchReleaseUrl = zipUrl.match(REGEX_ZIP_URL)
+        if (!matchReleaseUrl) {
+          throw new Error('The zipUrl (' + versionInZipUrl + ') is a invalid release URL')
```
Comment (jenslind) [inline src/GhReleases.js]: `versionInZipUrl` should be defined above.

**Comment by jenslind** [comment]
Nice, thanks! :cat2:
The `standard` lint is currently broken on travis builds, fixed it on master.
Also added some inline comments.

File: src/GhReleases.js (hunk 2 above)
Comment (maxclaus) [inline src/GhReleases.js]: Sorry I didn't get this one. What do you mean exactly?

File: src/GhReleases.js (hunk 1 above)
Comment (maxclaus) [inline src/GhReleases.js]: Sorry. Copy and paste :)

File: src/GhReleases.js (hunk 3 above)
Comment (maxclaus) [inline src/GhReleases.js]: Actually I wanted to show the `zipUrl`.

File: src/GhReleases.js (hunk 2 above)
Comment (jenslind) [inline src/GhReleases.js]: Rename the var `currentVersion` to `latestVersion`. :)

**Comment by jenslind** [comment]
`2.0.2`

**Pull request closed without merging**